src/fourier.jl

```diff
@@ -0,0 +1,773 @@
+        dRe = dIm = dG = @. NaNMath.sqrt(power_over_2n * (seg_power - frac))
+        # Eq. 19
+        dphi = @. NaNMath.sqrt(power_over_2n * (seg_power / (Gsq - bsq) -
+                       1 / (ref_power - ref_power_noise)))
+
+    else
+        PrPs = ref_power .* seg_power
+        dRe = @. NaNMath.sqrt((PrPs + real(cross_power) ^ 2 - imag(cross_power) ^ 2) /
+                      two_n_ave)
+        dIm = @. NaNMath.sqrt((PrPs - real(cross_power) ^ 2 + imag(cross_power) ^ 2) /
+                      two_n_ave)
+        gsq = raw_coherence.(cross_power, seg_power, ref_power,
+                            seg_power_noise, ref_power_noise, n_ave)
+        dphi = @. NaNMath.sqrt((1 - gsq) / (2 * gsq * n_ave))
+        dG = sqrt.(PrPs ./ n_ave)
+    end
+
+    return dRe, dIm, dphi, dG
+end
+
+function cross_to_covariance(cross_power::Complex, ref_power::Real, 
+                             ref_power_noise::Real, delta_nu::Real)
+    # To be used with dot broadcast when need an array 
+    return cross_power * sqrt(delta_nu / (ref_power - ref_power_noise))
+end
+
+function _which_segment_idx_fun(;binned::Bool=false, dt=nothing)
+    if binned
+        return (args...) -> generate_indices_of_segment_boundaries_binned(args...; dt=dt)
+    else
+        # Define a new function, so that we can pass the correct dt as an
+        # argument.
+        return generate_indices_of_segment_boundaries_unbinned 
+    end
+end
+
+function get_average_ctrate(times:: AbstractVector{<:Real}, gti::AbstractMatrix{<:Real}, 
+                            segment_size::Real; counts= nothing)
+    n_ph = 0.0
+    n_intvs = 0
+    binned = !isnothing(counts)
+    func = _which_segment_idx_fun(;binned)
+
+    for (_, _, idx0, idx1) in func(times, gti, segment_size)
+        if !(binned)
+            n_ph += idx1 - idx0
+        else
+            n_ph += sum(counts[idx0+1:idx1])
+        end
+        n_intvs += 1
+    end
+    return (n_ph / (n_intvs * segment_size))
+end
+
+@resumable function get_flux_iterable_from_segments(times:: AbstractVector{<:Real}, 
+                                                    gti::AbstractMatrix{<:Real}, 
+                                                    segment_size::Real; n_bin=nothing,
+                                                    fluxes=nothing, errors=nothing)
+    dt = nothing
+    binned = !isnothing(fluxes)
+    if binned
+        dt = Statistics.median(diff(times[1:100]))
+    end
+    fun = _which_segment_idx_fun(;binned, dt)
+
+    for (s, e, idx0, idx1) in fun(times, gti, segment_size)
+        if idx1 - idx0 < 2
+            @yield nothing
+            continue
+        end
+        if !binned
+            event_times = times[idx0:idx1-1]
+            # astype here serves to avoid integer rounding issues in Windows,
+            # where long is a 32-bit integer.
+            cts = fit(Histogram,Float64.(event_times .- s);nbins=n_bin).weights
+        else
+            cts = Float64.(fluxes[idx0+1:idx1])
```

Comment:
```suggestion
            event_times = @view times[idx0:idx1-1]
            # astype here serves to avoid integer rounding issues in Windows,
            # where long is a 32-bit integer.
            cts = fit(Histogram, float.(event_times .- s); nbins=n_bin).weights
        else
            cts = float.(@view fluxes[idx0+1:idx1])
```